rclc/include/rclc/types.h

```diff
@@ -21,20 +21,19 @@
 #include "rcl/types.h"
 
 typedef rcl_ret_t rclc_ret_t;
-typedef rcl_node_t rclc_node_t;
-typedef rcl_publisher_t rclc_publisher_t;
 
-typedef void (* rclc_callback_t)(const void *);
+struct rclc_node_t;
+typedef struct rclc_node_t rclc_node_t;
+
+struct rclc_publisher_t;
+typedef struct rclc_publisher_t rclc_publisher_t;
 
-typedef struct rclc_subscription_t
-{
-  struct rclc_subscription_t * rcl_subscription;
-  rclc_callback_t user_callback;
-} rclc_subscription_t;
+struct rclc_subscription_t;
+typedef struct rclc_subscription_t rclc_subscription_t;
 
-typedef struct rclc_executor_t
-{
-  // TODO(wjwwood): Fill this out
-} rclc_executor_t;
+struct rclc_executor_t;
+typedef struct rclc_executor_t rclc_executor_t;
+
+typedef void (* rclc_callback_t)(const void *);
```

Comment:
Well, it can be a big discussion, but I don't like giving to the user the internal implementation of my structs (the user should only use the functions to manipulate these structs). To my opinion, only the things that are useful for the user should be in the `include` directory.  Also, I usually do that when the memory allocation is done by the library and not by the user. If some definition is useful for several sources files of the library, I would just put a header file in the `src` directory.